firebase-inappmessaging-display/src/main/java/com/google/firebase/inappmessaging/display/FirebaseInAppMessagingDisplay.java

```diff
@@ -236,11 +221,42 @@ public void onActivityDestroyed(Activity activity) {
   @Override
   public void onActivityResumed(Activity activity) {
     super.onActivityResumed(activity);
+    bindFiamToActivity(activity);
+  }
+
+  private void bindFiamToActivity(Activity activity) {
+    if (!currentActivityName.equals(activity.getLocalClassName())) {
+      Logging.logi("Binding to activity: " + activity.getLocalClassName());
+      headlessInAppMessaging.setMessageDisplayComponent(
+          (iam, cb) -> {
+            // When we are in the middle of showing a message, we ignore other notifications these
+            // messages will be fired when the corresponding events happen the next time.
+            if (inAppMessage != null || headlessInAppMessaging.areMessagesSuppressed()) {
+              Logging.logd("Active FIAM exists. Skipping trigger");
+              return;
+            }
+            inAppMessage = iam;
+            callbacks = cb;
+            showActiveFiam(activity);
+          });
+      // set the current activity to be the one passed in
+      currentActivityName = activity.getLocalClassName();
+    }
     if (inAppMessage != null) {
       showActiveFiam(activity);
     }
   }
 
+  private void unbindFiamFromActivity(Activity activity) {
+    if (currentActivityName.equals(activity.getLocalClassName())) {
+      Logging.logi("Unbinding from activity: " + activity.getLocalClassName());
+      headlessInAppMessaging.clearDisplayListener();
+      imageLoader.cancelTag(activity.getClass());
+      removeDisplayedFiam(activity);
+      currentActivityName = "";
```

Comment:
Should we set this to null and add `@Nullable` annotation on this variable? I think an empty name can be error prone